### PR TITLE
test(async): cover async server HTTP endpoints (tests only)

### PR DESCRIPTION
**Pull request by endolith**
## Background

`interpreter/core/async_core.py` (the async HTTP/WebSocket server) is the largest under-covered file in the repo (~45% statement coverage). The recently fixed send_output drain loop lives here, and CI's codecov/project gate flags regressions on it.

## Problem

The HTTP endpoints had no coverage. This PR is **tests only** — it documents the current behavior of every endpoint, including several that look wrong.

## What this PR changes

- **`tests/core/test_async_server_endpoints.py`** (new, 29 tests): `/heartbeat`, `/` (GET + POST), `/settings` GET + error branches, the OpenAI-compatible `/openai/chat/completions` endpoint (sentinels, context-mode/auto-run toggles, text/list/base64-image content, run-code streaming, confirmation-break streaming), and the opt-in insecure `/run`, `/upload`, `/download` routes (registered only when `INTERPRETER_INSECURE_ROUTES=true`).
- **`tests/core/test_async_core_extended.py`**: respond() error/retry branches and accumulate() bytes-content branches.

The buggy-looking behaviors are **documented, not fixed**, each marked `KNOWN BUG`:
- Flask-style `(content, status)` tuple returns that FastAPI serializes as HTTP 200 JSON arrays (never set a status code).
- `POST /` never awaits `input()`, so the payload is never processed and the error path is dead code.
- `/download/{filename}` cannot match paths with slashes.

These are resolved in a separate follow-up PR that flips the tests to the corrected behavior.

## Tests

- `pytest -m "not integration"` → 807 passed against **un-fixed** `main` code (all tests pass because they document current behavior).
- `ruff check` clean.

## Summary by CodeRabbit

* **Tests**
  * Expanded coverage for asynchronous response handling, including byte accumulation, generator failures, retries, and empty responses.
  * Added comprehensive tests for server endpoints, including health checks, settings, chat interactions, control tokens, text and image inputs, streaming, and file operations.
  * Documented current edge-case behaviors in endpoint responses and downloads.

### CODE REVIEW

File: tests/core/test_async_core_extended.py [hunk 1]
```diff
@@ -77,3 +77,84 @@ def test_input_end_chunk_with_queued_stop_command_joins_response_thread():
...
+    async_i.auto_run = True
+
+    def empty_generator():
+        if False:
+            yield  # pragma: no cover
+
+    with mock.patch.object(async_i, "_respond_and_store", empty_generator):
+        with mock.patch("interpreter.core.async_core.time.sleep"):
+            with pytest.raises(Exception, match="No chunks sent"):
+                async_i.respond()
```
[the remark this answers is not in the document]
Comment (endolith) [reply tests/core/test_async_core_extended.py:130]: Fixed in fe459fa3e0: the test now stubs _respond_and_store with a mock returning iter(()) and asserts call_count == 5, verifying the retry loop actually retries five times. — [opencode](https://github.com/anomalyco/opencode)
— [opencode](https://github.com/anomalyco/opencode)

File: tests/core/test_async_core_extended.py [hunk 2]
```diff
@@ -77,3 +77,84 @@ def test_input_end_chunk_with_queued_stop_command_joins_response_thread():
...
+
+def test_respond_emits_error_when_no_generator_defined():
+    """respond() must put an error message when the interpreter has no messages to respond to."""
+    async_i = AsyncInterpreter()
+    mock_q = mock.MagicMock()
+    async_i.output_queue = mock.MagicMock(sync_q=mock_q)
+    async_i.auto_run = True
+    async_i.messages = []
+
+    with mock.patch.object(async_i, "_respond_and_store", return_value=[]):
```
[the remark this answers is not in the document]
Comment (endolith) [reply tests/core/test_async_core_extended.py]: Fixed in fe459fa3e0: renamed to test_respond_emits_error_when_response_is_empty with a matching docstring, and dropped the irrelevant async_i.messages = [] line. — [opencode](https://github.com/anomalyco/opencode)
— [opencode](https://github.com/anomalyco/opencode)

File: tests/core/test_async_server_endpoints.py [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+    KNOWN BUG: the handler calls async_interpreter.input(payload) without
+    await, so the coroutine is never awaited and its exception is never
+    raised. It responds 200 "success" and the error path is dead code. This
+    test documents the current (wrong) behavior.
+    """
+    _, interpreter = server_pair
+    interpreter.input = mock.AsyncMock(side_effect=ValueError("boom"))
+    response = client.post("/", json={"role": "user", "type": "message", "start": True})
+    assert response.status_code == 200
+    assert response.json() == {"status": "success"}
```
[the remark this answers is not in the document]
Comment (endolith) [reply tests/core/test_async_server_endpoints.py:79]: This is the `POST /` unawaited-input bug. Per the split decision, bug fixes live in the dedicated follow-up PR #242 (fix/async-server-endpoint-bugs), which awaits `input(payload)`, returns the proper FastAPI 500 error response, and flips this test to assert `response.status_code == 500` with `{"error": "boom"}`. No change here; resolved when #242 lands. — [opencode](https://github.com/anomalyco/opencode)

File: tests/core/test_async_server_endpoints.py [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+
+def test_post_settings_unknown_top_level_key_returns_200_with_tuple_body(client):
+    """POST /settings with an unknown top-level key returns 200 with an array body.
+
+    KNOWN BUG: same Flask-style tuple issue; should be a 404 response but is
+    HTTP 200 with a JSON array [..., 404].
+    """
+    response = client.post("/settings", json={"no_such_setting": True})
+    assert response.status_code == 200
+    assert response.json() == [{"error": "Setting no_such_setting not found"}, 404]
```
[the remark this answers is not in the document]
Comment (endolith) [reply tests/core/test_async_server_endpoints.py:123]: This is the Flask-style tuple / status-code bug. Per the split decision, the fix lives in #242 (fix/async-server-endpoint-bugs), which converts all error tuple returns to JSONResponse(..., status_code=...) and flips these tests to assert the real 400/404/500 statuses and error-object bodies. No change here; resolved when #242 lands.
— [opencode](https://github.com/anomalyco/opencode)

File: tests/core/test_async_server_endpoints.py [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+    """The /download route does not match paths containing slashes.
+
+    KNOWN BUG: the route is declared as /download/{filename}, which matches a
+    single path segment. Any path with a slash (including the absolute paths
+    the endpoint is meant to serve) falls through to 404. Documenting current
+    behavior; a fix would use {filename:path}.
+    """
+    client, _ = insecure_pair
+    response = client.get("/download/some/dir/file.bin")
+    assert response.status_code == 404
```
[the remark this answers is not in the document]
Comment (endolith) [reply tests/core/test_async_server_endpoints.py:444]: This is the /download single-segment route bug. Per the split decision, the fix lives in #242 (fix/async-server-endpoint-bugs), which changes the route to /download/{filename:path} and flips this test to download a real slashed file path (tmp_path). No change here; resolved when #242 lands.
— [opencode](https://github.com/anomalyco/opencode)

**Comment by endolith** [comment]
@coderabbit review
